lib/util/middleware.js

```diff
@@ -0,0 +1,64 @@
+if (typeof (fetch) === 'undefined') require('isomorphic-fetch')
+
+const API_USER_PATH = '/api/secure/user'
+
+export async function secureFetch (url, accessToken, apiKey, method = 'get', options = {}) {
+  const res = await fetch(url, {
+    method,
+    mode: 'cors', // Middleware is at a different URL.
+    headers: {
+      Authorization: `Bearer ${accessToken}`,
+      'Content-Type': 'application/json',
+      'x-api-key': apiKey
+    },
+    ...options
+  })
+  if ((res.status && res.status >= 400) || (res.code && res.code >= 400)) {
+    const result = await res.json()
+    let message = `Error ${method}-ing user: ${result.message}`
+    if (result.detail) message += `  (${result.detail})`
+
+    console.log(message)
+    return {
+      status: 'error',
+      message
+    }
+  }
+  return {
+    status: 'success',
+    data: await res.json()
+  }
+}
+
+export async function fetchUser (middlewareConfig, idField, id, token) {
+  const { apiBaseUrl, apiKey } = middlewareConfig
+  const requestUrl = `${apiBaseUrl}${API_USER_PATH}/find/${idField}/${id}`
+
+  return secureFetch(requestUrl, token, apiKey)
+}
+
+export async function addUser (middlewareConfig, token, data) {
```

Comment:
It really seems like we would want these to be actions that update the redux store when they receive the response.

Reply:
Yes... coming in the next PR. The next PR will make the current user a redux state!